Support GitHub Releases

roynos opened the pull request:
#### c3408c8ba77a0bbcfbc7aa1dfd6f84516e9320f0
<pre>
Support GitHub Releases
<a href="https://bugs.webkit.org/show_bug.cgi?id=244910">https://bugs.webkit.org/show_bug.cgi?id=244910</a>
rdar://99667302

Reviewed by Jonathan Bedard.

* Tools/Scripts/libraries/webkitscmpy/setup.py:
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/remote/git_hub.py:

Canonical link: <a href="https://commits.webkit.org/254286@main">https://commits.webkit.org/254286@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5ac48aa42e7705d198367928543c38f5fe85530a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/88644 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/33206 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/19527 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/97833 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/154397 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/92640 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/31698 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/27291 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/80865 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/92469 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/94275 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/68/builds/25146 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/75603 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/25088 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [❌ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/92235 "Found 1 webkitpy python2 test failure: webkitcorepy.tests.file_lock_unittest.FileLockTestCase.test_locked_timeout") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/80030 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/80096 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/68052 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/29395 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/14095 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/87703 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/29224 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/15108 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3030 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/32670 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/75603 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/31353 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/34219 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/Scripts/libraries/webkitscmpy/webkitscmpy/remote/git_hub.py:
```diff
@@ -345,19 +345,22 @@ def request(self, path=None, params=None, headers=None, authenticated=None, pagi
+            params['per_page'] = params.get('per_page', 100)
+            params['page'] = params.get('page', 1)
+
+        json = {key: value for key, value in json.items()} if json else dict()
```
Shouldn't this just be "None" if no json is defined? Otherwise we'll send an empty dictionary to on every `GET` request?

roynos replied:
Done.

JonWBedard commented on Tools/Scripts/libraries/webkitscmpy/webkitscmpy/remote/git_hub.py:
```diff
@@ -694,3 +697,14 @@ def files_changed(self, argument=None):
             if file.get('filename')
         ]
+
+    def releases(self, name, tag_name, target_commitish='main', body='', draft=False, prerelease=False):
```
Does this create or list releases? `releases` would indicate it just lists releases, but this looks like it _creates_ them? Should probably name `create_release` or something to that effect.

JonWBedard commented on Tools/Scripts/libraries/webkitscmpy/setup.py:
```diff
@@ -29,7 +29,7 @@ def readme():
 setup(
     name='webkitscmpy',
-    version='5.6.1',
+    version='5.6.2',
```
Need to modify `Tools/Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py` too

roynos committed:
Support GitHub Releases
https://bugs.webkit.org/show_bug.cgi?id=244910
rdar://99667302

Reviewed by Jonathan Bedard.

* Tools/Scripts/libraries/webkitscmpy/setup.py:
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/remote/git_hub.py:

Canonical link: https://commits.webkit.org/254286@main
